### PR TITLE
[ADD} connect create microsite screens to backend

### PR DESCRIPTION
**Pull request by ekediala**
Added the functionality for sending microsite data to backend.

Before testing, please request updated `app.json`.

### CODE REVIEW

File: src/components/createMicrosite/components/ThirdStage.jsx [hunk 1]
```diff
@@ -35,11 +40,11 @@ const {
 
 const ThirdStage = ({ prev }) => {
     const state = useSelector(state => state.microsite);
-    const { photoURL, categories, tags } = state;
+    // const uid = useSelector(state => state.login.user.uid);
+    const uid = 'zSHCoHu10lgYysuMFvbG6dkXUw82';
```
Comment (mrayor) [inline src/components/createMicrosite/components/ThirdStage.jsx]: what is happening here?

File: src/components/createMicrosite/components/ThirdStage.jsx [hunk 2]
```diff
@@ -51,13 +56,20 @@ const ThirdStage = ({ prev }) => {
...
     }, [imgSrc, chosenCategories]);
 
     const finish = () => {
-        const storeCreationAction = createStoreAction();
+        const loadingAction = createLoadingAction(true);
+        dispatch(loadingAction);
+        // if (photoURL.uri !== DEFAULT_IMG_URI) {
+        //     const uploadImageAction = createUploadImageAction(photoURL);
+        //     dispatch(uploadImageAction);
+        // }
```
Comment (mrayor) [inline src/components/createMicrosite/components/ThirdStage.jsx]: why the commented code?

File: src/components/createMicrosite/constants.js [hunk 3]
```diff
@@ -174,3 +175,53 @@ export const DEFAULT_LOCATION = {
...
+    {
+        description: 'Everything Software',
+        key: 'hhhjkdhdhd',
+        name: 'Men',
+    },
+
+    {
+        description: 'Everything Software',
+        key: 'tegpoovd',
+        name: 'Kids',
```
Comment (mrayor) [inline src/components/createMicrosite/constants.js]: I know its a dummy but at least use better keys

File: src/components/createMicrosite/helpers.js [hunk 4]
```diff
@@ -8,3 +8,29 @@ export const getImageSrc = async setlogoSrc => {
...
+    }
+    if (description === undefined) {
+        toast.showToast('Business description required');
+        return false;
+    }
+    if (categories.length < 1) {
+        toast.showToast('Please select at least one category');
+        return false;
+    }
+    return true;
```
Comment (mrayor) [inline src/components/createMicrosite/helpers.js]: We would fix this validation with Formik/Yup

File: src/components/apis/sb.js [hunk 5]
```diff
@@ -1,5 +1,9 @@
+import Constants from 'expo-constants';
 import axios from 'axios';
 
+const { bearer } = Constants.manifest.extra.axios;
+
 export default axios.create({
     baseURL: 'https://us-central1-space-barter.cloudfunctions.net/api/v1',
```
Comment (parkerthegeniuschild) [inline src/components/apis/sb.js]: if you added the Bearer token to a constant, i see no reason why the apiURL can not be added to . . .

File: src/components/createMicrosite/components/ThirdStage.jsx (hunk 2 above)
Comment (ekediala) [reply src/components/createMicrosite/components/ThirdStage.jsx]: Uploading images is taking too long. Has to be moved to next sprint but that code handles it.

File: src/components/createMicrosite/constants.js (hunk 3 above)
Comment (ekediala) [reply src/components/createMicrosite/constants.js]: Lol. Would be deleted. Why bother?

File: src/components/createMicrosite/helpers.js (hunk 4 above)
Comment (ekediala) [reply src/components/createMicrosite/helpers.js]: Tried Yup. It was throwing weird errors.

File: src/components/createMicrosite/components/ThirdStage.jsx (hunk 1 above)
Comment (ekediala) [reply src/components/createMicrosite/components/ThirdStage.jsx]: There is no user available at the time to test so I pulled a registered user to test. Forgot to uncomment that line.

**Comment by parkerthegeniuschild** [comment]
The codes are nothing short of impressive but i do have issues with the user interface and much more, the user experience. This is a critical review; fix what can be within the allotted time frame.

1. I don't understand why you are adding key and id to tags, just the name and description would so when it is associated with a business. The key and id are for the GET request itself.
---
image
---
2. You already have a Create Microsite at the very top. there is really no need to say FILL IN MICROSITE DATA again . . especially printing it in all caps . . . some users find all caps pretty annoying . . .as they feel it's synonymous to screaming. And yes it does look like you were, cos you keep smashing it everyone . . 

3. Categories don't work efficiently, I tap the button and expect to see a drop drown list . .  but nothing happens, until I start typing . . Which means I am forced to type my own opinions and if it exists then fine. If i believe my biz category is called: 'Hollandia Milk", now we have a problem.

4. Subtle grammar errors . . . `Categories>`

5. Business location . . Open Map does nothing but open up a map and I have to be outstanding with geography to read and understand that map i.e. start dragging, pinching and zooming to get my location. i figure it would take hours to pinch and zoom all the way to Abuja if i am in Lagos.   Too many options too . . it gets boring. While the other option seem to work, perhaps, cos there's no feedback of any sort, when i choose any of the buttons, neither is an address displayed to assure me I did the right thing . . they don't work . . .cos a quick look at the database and I cannot find any location entry.

This is a lot.

File: src/components/createMicrosite/components/ThirdStage.jsx (hunk 1 above)
Comment (parkerthegeniuschild) [reply src/components/createMicrosite/components/ThirdStage.jsx]: remove `commented out codes` from production ready prs

File: src/components/createMicrosite/components/ThirdStage.jsx (hunk 2 above)
Comment (parkerthegeniuschild) [reply src/components/createMicrosite/components/ThirdStage.jsx]: remove commented out codes from production-ready prs

File: src/components/createMicrosite/constants.js (hunk 3 above)
Comment (parkerthegeniuschild) [reply src/components/createMicrosite/constants.js]: LOL

**Comment by ekediala** [comment]
Alright. Will fix as much as I can within the allotted time.

**Pull request merged**